### PR TITLE
4840-Cypress-Login-Tests

### PR DESCRIPTION
**Pull request by androolloyd**
##### Description

Global Cypress changes
 - config is located in cypress/config/development.json, setup environment variables here
 - cypress plugins extended to support the promise workflow outlined here : https://docs.cypress.io/api/plugins/configuration-api.html\#Usage

Cypress Commands Added:
 - login - 1 argument: userType (funder|hunter)
 - persona - 2 arguments: userType, forcePersona = false

Login Tests added:
 - funder/hunter workflow
 - NEW_USER env variable is for when there is no user in the database, only works once presently need to add destructive scripts to setup db state

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/4840

##### Testing

Videos are not available to upload directly apparently: 
https://gitcoincore.slack.com/files/ULDQM8EFK/FLU4HQVQW/automated_funder_demo_login.mov

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@androolloyd that PR description AND CYPRESSS \m/ ❤️ 
Is this still a WIP -> could we make it draft PR ?

**Comment by androolloyd** [comment]
Not entirely sure why travis is having a fit, I don't see any errors in its attempt to build.

**Comment by thelostone-mc** [comment]
@androolloyd seems we've got lint errors!
Could you run `npm run eslint` ? 

We've got pre-commit hooks set up with the repo which run lint checks + isort
You can install that and the checks kick in each time you create a local commit ^_^

**Comment by androolloyd** [comment]
I screwed up my local.env, thinking I just stuck it there by accident when in fact its apart of the doc process to copy that local.env into the .env file location in app/app.

I updated the .gitignore to capture **/**/*.env and *.env to ensure that regardless of file depth its not going to allow .env changes to occur.

File: .gitignore [hunk 1]
```diff
@@ -25,7 +25,8 @@ webpack-stats.json
 *.sql
 
 # Sensitive environment files
-.env
+**/**/.env
+*.env
```
Comment (octavioamu) [inline .gitignore]: I think this will remove local.env needed for the first steps setup on our docs https://docs.gitcoin.co/mk_setup/

**Comment by thelostone-mc** [comment]
@danlipert / @octavioamu this look good to get in ?

**Comment by danlipert** [comment]
@thelostone-mc I need to set some time aside to manually check it out first, maybe over the holidays

**Comment by thelostone-mc** [comment]
Yo closing this out for now but will reopen when we decide to to get around to writing tests

**Pull request closed without merging**